emlparser/emlparser.py

```diff
@@ -145,10 +145,8 @@ def execute(self, request):
             content_str = html_email.as_bytes()
 
         # Verify this is truly an eml file before attempting parsing.
-        # Assumption: contents should be able to decode to UTF-8
-        try:
-            content_str.decode()
-        except UnicodeDecodeError:
+        # Assumption: Contents should contain no null bytes
+        if b'\x00' in content_str:
```

Comment:
what if they do though? shouldn't we just replace them with ""?

Reply:
We could, the idea is to filter out potential scripts that got identified as document/email (probably because the presence of header fields like From, To, etc.).

After going through the different types of samples (eml, msg, msg + .exe attachment, eml w/ VISCII), this seems the best condition but I'm open to suggestions 😀

Reply:
Is it possible for a script that got identified as document/email, contains null bytes, and is valid? If so then we should cut the null bytes

Reply:
Well if it's a script at all, it wouldn't be valid since the service can only parse eml files (either directly or by conversion).

So these kind of scripts shouldn't be allowed to be parsed by the service.

Reply:
I agree, but I don't think the presence of null bytes guarantees that the file is a script and not an eml file

Reply:
Replacing null bytes at the very least prevents the service for getting stuck, we'll go with that until more errors appear to narrow this down.